core/modules/system/system.install

```diff
@@ -589,13 +589,13 @@ function system_requirements($phase) {
 
   if ($phase == 'runtime') {
     // Inform the user if theme_debug is enabled.
-    $theme_debug_enabled = config_get('system.core', 'theme_debug');
+    $theme_debug_enabled = settings_get('theme_debug', config_get('system.core', 'theme_debug'));
```

Comment:
In the spirit of not having huge chunks of help text, I think that the bit about settings.php should be added conditionally. We have a way to figure out where the setting is actually set, so no point in sending users chasing for it in various places. So, perhaps check if the setting has been set in config vs settings php, and show the appropriate message to users. Something like this:

```suggestion
    $theme_debug_enabled = settings_get('theme_debug', config_get('system.core', 'theme_debug'));
    $theme_debug_description = $t('Theme debugging is currently enabled, and should be disabled on a live site.);
    if (settings_get('theme_debug') != NULL) {
      $theme_debug_description .= ' ' . $t('This is currently enabled in your <code>settings.php</code> file.');
    }
    elseif (config_get('system.core', 'theme_debug') != NULL) {
      $theme_debug_description .=  ' ' . $t('This setting can be changed using the <a href="!url">Devel module</a>', array('!url' => module_exists('devel') ? url('admin/config/development/devel') : 'https://backdropcms.org/project/devel'));
    }
```

PS: I would really like us to expose the `theme_debug` setting in the admin UI (somewhere under Configuration -> Development), instead of asking people to install a contrib module to achieve that. Separate issue though 😅